manifests/check.pp

```diff
@@ -107,11 +107,18 @@
 #   Set this to 'absent' to remove it completely.
 #   Default: undef
 #
+# [*content*]
```

Comment:
> It was my understanding that custom adds key/values at top scope. So if you want data from custom to be in the checks scope you need the hash in custom specified as such.

We could change the behavior to operate like that, but today the value of custom is assumed to be a hash merged with the value of checks.mycheck.  Check it out at https://github.com/sensu/sensu-puppet/blob/v2.30.0/lib/puppet/provider/sensu_check/json.rb#L92

> Would that work? Does Puppet's $hash + $another_hash work for that?

We'd probably want to use the deep_merge() function in stdlib.  The `+` operator isn't a deep merge, so duplicated keys result in the rightmost value clobbering the other values.

An example of the gymnastics I'm doing with the Puppet DSL to merge nested hash values is at https://github.com/sensu/sensu-puppet/pull/785/commits/cddda1a07d78c2699f10b07fbfb2c5376358311e#diff-7bfaf46cbbb09c017062dc6fbdfdd383R262